app/src/test/java/org/mozilla/fenix/library/history/HistoryInteractorTest.kt

```diff
@@ -0,0 +1,157 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+   License, v. 2.0. If a copy of the MPL was not distributed with this
+   file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.library.history
+
+import io.mockk.mockk
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class HistoryInteractorTest {
+
+    @Test
+    fun onHistoryItemOpened() {
+        var historyItemReceived: HistoryItem? = null
+        val historyItem = HistoryItem(0, "title", "url", 0.toLong())
+        val interactor = HistoryInteractor(
+            mockk(),
+            { historyItemReceived = it },
+            mockk(),
+            mockk(),
+            mockk()
+        )
+        interactor.onHistoryItemOpened(historyItem)
+        assertEquals(historyItem, historyItemReceived)
+    }
+
+    @Test
+    fun onEnterEditMode() {
+        val store: HistoryStore =
+            HistoryStore(HistoryState(listOf(), HistoryState.Mode.Normal), ::historyStateReducer)
+        val interactor =
+            HistoryInteractor(store, mockk(), mockk(), mockk(), mockk())
+        val historyItem = HistoryItem(0, "title", "url", 0.toLong())
+        runBlocking {
+            interactor.onEnterEditMode(historyItem)
+            delay(50)
```

Comment:
For the `SearchFragment`, we structured the tests to avoid the coroutine scope in the interactor test, which also helped us get rid of the `delay` (as it might cause intermittently failing tests).

In the `SearchInteractorTest`, we just verify that the right actions are dispatched: https://github.com/mozilla-mobile/fenix/blob/master/app/src/test/java/org/mozilla/fenix/search/SearchInteractorTest.kt

Then we added the `SearchStoreTest` to make sure all actions have the intended effect: https://github.com/mozilla-mobile/fenix/blob/master/app/src/test/java/org/mozilla/fenix/search/SearchStoreTest.kt

Let's follow the same pattern here?

Reply:
Done! I *think* the tests should be correct now

Reply:
Tests look great now!